kong/runloop/plugins_iterator.lua

```diff
@@ -169,55 +182,36 @@ end
 -- @tparam string|nil consumer_id The consumer identifier.
 -- @return any|nil The configuration corresponding to the best matching combination, or 'nil' if no configuration is found.
 ---
-function PluginsIterator.lookup_cfg(combos, route_id, service_id, consumer_id)
-  -- Use the build_compound_key function to create an index for the 'combos' table
-  local build_compound_key = PluginsIterator.build_compound_key
-
-    local key
-    if route_id and service_id and consumer_id then
-        key = build_compound_key(route_id, service_id, consumer_id)
-        if combos[key] then
-            return combos[key]
-        end
-    end
-    if route_id and consumer_id then
-        key = build_compound_key(route_id, nil, consumer_id)
-        if combos[key] then
-            return combos[key]
-        end
-    end
-    if service_id and consumer_id then
-        key = build_compound_key(nil, service_id, consumer_id)
-        if combos[key] then
-            return combos[key]
-        end
-    end
-    if route_id and service_id then
-        key = build_compound_key(route_id, service_id, nil)
-        if combos[key] then
-            return combos[key]
-        end
-    end
-    if consumer_id then
-        key = build_compound_key(nil, nil, consumer_id)
-        if combos[key] then
-            return combos[key]
-        end
-    end
-    if route_id then
-        key = build_compound_key(route_id, nil, nil)
-        if combos[key] then
-            return combos[key]
-        end
+local function lookup_cfg(combos, route_id, service_id, consumer_id)
+  local ids = PLUGIN_IDS
+
+  ids[1] = route_id
+  ids[2] = service_id
+  ids[3] = consumer_id
+
+  for i = 1, #PLUGIN_ORDERS do
```

Comment:
Below you changed `3` to `PLUGIN_ORDER_N`. It you go that route, this could also be changed to constant `PLUGIN_ORDERS_N = #PLUGIN_ORDERS`.